fv3core/stencils/updatedzc.py

```diff
@@ -1,263 +1,98 @@
-    fy = yfx * (gz_y[0, -1, 0] if yfx > 0.0 else gz_y)
-    return fx, fy
-
-
-#      do j=js1, je1
-#         do i=is1,ie1
-#            gz(i,j,k) = (gz2(i,j)*area(i,j) +  fx(i,j)- fx(i+1,j)+ fy(i,j)- fy(i,j+1)) & # noqa: E501
-#                      / (         area(i,j) + xfx(i,j)-xfx(i+1,j)+yfx(i,j)-yfx(i,j+1)) # noqa: E501
-#         enddo
-#      enddo
-
-#   end do
-
-# ! Enforce monotonicity of height to prevent blowup
-#   do i=is1, ie1
-#      do j=js1, je1
-#         ws(i,j) = ( zs(i,j) - gz(i,j,km+1) ) * rdt
-#      enddo
-#   end do
-
-#   do k = km, 1, -1
-#      do j=js1, je1
-#         do i=is1, ie1
-#            gz(i,j,k) = max( gz(i,j,k), gz(i,j,k+1) + dz_min )
-#         enddo
-#      enddo
-#   enddo
-
-
 @gtstencil()
-def update_dz_c(
-    dp_ref: sd,
-    zs: sd,
-    area: sd,
-    ut: sd,
-    vt: sd,
-    gz: sd,
-    gz_x: sd,
-    gz_y: sd,
-    ws3: sd,
-    *,
-    dt: float,
+def update_dz_c_stencil(
+    area: FloatField,
+    dp_ref: FloatField,
+    gz_surface: FloatField,
+    ut: FloatField,
+    vt: FloatField,
+    gz: FloatField,
+    ws: FloatField,
+    dt2: float,
 ):
+    """Update the model heights from the C-grid wind flux
+
+    After the model runs c_sw and advances the c-grid variables half a timestep,
+    the grid deforms with the flow in the Lagrangian coordinate system. This
+    module updates the model heights based on the flux, and the rate of change of
+    the lowest model height compared to the fixed surface height.
+
+    Args:
+         dp_ref: vertical difference in column reference pressure (in)
+         gz_surface: surface geopotential height (m) (in)
+         ut: x-velocity on the C-grid, contravariant of the D-grid winds (in)
+         vt: y-velocity on the C-grid, contravariant of the D-grid winds (in)
+         gz: geopotential height of the model grid cells (m) (inout)
+         ws: rate of change in the surface geopotential height from the C-grid
+             wind (inout)
+         dt2: timestep of the C-grid update in seconds (in)
+    Grid variable inputs:
+         area
+    """
+    from __externals__ import local_ie, local_is, local_je, local_js
+
     with computation(PARALLEL):
         with interval(0, 1):
-            xfx = p_weighted_average_top(ut, dp_ref)
-            yfx = p_weighted_average_top(vt, dp_ref)
+            u_average = p_weighted_average_top(ut, dp_ref)
+            v_average = p_weighted_average_top(vt, dp_ref)
         with interval(1, -1):
-            xfx = p_weighted_average_domain(ut, dp_ref)
-            yfx = p_weighted_average_domain(vt, dp_ref)
+            u_average = p_weighted_average_domain(ut, dp_ref)
+            v_average = p_weighted_average_domain(vt, dp_ref)
         with interval(-1, None):
-            xfx = p_weighted_average_bottom(ut, dp_ref)
-            yfx = p_weighted_average_bottom(vt, dp_ref)
+            u_average = p_weighted_average_bottom(ut, dp_ref)
+            v_average = p_weighted_average_bottom(vt, dp_ref)
```

Comment:
Woooah I actually thought these were pressure for some reason, thanks for updating the names!